File: java/BOJ/src/boj10828.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class boj10828 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        StringTokenizer stk;

        int[] stack = new int[n];
        int size = 0;

        for (int i = 0; i < n; i++) {
            stk = new StringTokenizer(br.readLine(), " ");
            String order = stk.nextToken();
            switch (order) {

                case "push":
                    stack[size] = Integer.parseInt(stk.nextToken());
                    size++;
                    break;

                case "pop":
                    if (size == 0) {
                        System.out.println(-1);
                        continue;
                    }
                    size--;
                    System.out.println(stack[size]);
                    break;

                case "size":
                    System.out.println(size);
                    break;

                case "empty":
                    if (size == 0) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;
                case "top":
                    if (size == 0) {
                        System.out.println(-1);
                    } else {
                        System.out.println(stack[size-1]);
                    }
            }
        }
    }
}

File: python/boj10828.py
# boj 10828 스택
import sys
input = sys.stdin.readline
n = int(input())

stack = []

for i in range(n):

    order = input().split()

    if order[0] == "push":
        stack.append(int(order[1]))
    elif order[0] == "top":
        if len(stack) == 0:
            print(-1)
            continue
        print(stack[-1])
    elif order[0] == "pop":
        if len(stack) == 0:
            print(-1)
            continue
        top = stack.pop()
        print(top)
    elif order[0] == "size":
        print(len(stack))
    elif order[0] == "empty":
        if len(stack) == 0:
            print(1)
        else:
            print(0)


